src/test/unit_tests/test_nc_iam_basic_integration.js

```diff
@@ -0,0 +1,234 @@
+    });
+
+    mocha.after(async () => {
+        fs_utils.folder_delete(`${config_root}`);
+    });
+
+    mocha.describe('IAM User API', async function() {
+        const username = 'Asahi';
+        const new_username = 'Botan';
+
+        mocha.it('list users - should be empty', async function() {
+            const input = {};
+            const command = new ListUsersCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+        });
+
+        mocha.it('create a user', async function() {
+            const input = {
+                UserName: username
+            };
+            const command = new CreateUserCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+            assert.equal(response.User.UserName, username);
+        });
+
+        mocha.it('list users - should have 1 user in the list', async function() {
+            const input = {};
+            const command = new ListUsersCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+            assert.equal(response.Users.length, 1);
+            assert.equal(response.Users[0].UserName, username);
+        });
+
+        mocha.it('get a user', async function() {
+            const input = {
+                UserName: username
+            };
+            const command = new GetUserCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+            assert.equal(response.User.UserName, username);
+        });
+
+        mocha.it('update a user', async function() {
+            const input = {
+                NewUserName: new_username,
+                UserName: username
+            };
+            const command = new UpdateUserCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+        });
+
+        mocha.it('list users - should have 1 user in the list (after updating the username)', async function() {
+            const input = {};
+            const command = new ListUsersCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+            assert.equal(response.Users.length, 1);
+            assert.equal(response.Users[0].UserName, new_username);
+        });
+
+        mocha.it('delete a user', async function() {
+            const input = {
+                UserName: new_username // delete a user after its username was updated
+            };
+            const command = new DeleteUserCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
```

Comment:
worth creating a function with assert.equal(response.$metadata.httpStatusCode, 200); instead of repeating it